legend-sdlc-generation-model/src/test/java/org/finos/legend/sdlc/generation/model/TestModelGenerationFactory.java

```diff
@@ -0,0 +1,166 @@
+        Class myClass = (Class) generatedModel.getElements().get(0);
+        Assert.assertEquals("MyClass", myClass.name);
+    }
+
+    @Test
+    public void testMultiStepGenerator()
+    {
+        GenerationSpecification generationSpecification = this.getTestGenerationSpecification();
+        PureModelContextData.Builder builder = PureModelContextData.newBuilder();
+        builder.addElement(generationSpecification);
+        ModelGenerationFactory factory = ModelGenerationFactory.newFactory(generationSpecification, builder.build());
+        factory.processModelGenerator(simpleClassGenerator);
+        factory.processModelGenerator(classDependedOnSimpleClassGenerator);
+        PureModel pureModel = factory.getPureModel();
+        org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.type.Class<?> _class = pureModel.getClass("model::MyComplexClass");
+        Assert.assertEquals(_class.getName(), "MyComplexClass");
+        Assert.assertEquals(pureModel.getModelClasses().size(), 2);
+        PureModelContextData generatedModel = factory.validateAndBuildGeneratedModel();
+        Assert.assertEquals(generatedModel.getElements().size(), 2);
+    }
+
+    @Test
+    public void testCompileError()
+    {
+        GenerationSpecification generationSpecification = this.getTestGenerationSpecification();
+        PureModelContextData.Builder builder = PureModelContextData.newBuilder();
+        builder.addElement(generationSpecification);
+        ModelGenerationFactory factory = ModelGenerationFactory.newFactory(generationSpecification, builder.build());
+        try
+        {
+            factory.processModelGenerator(classDependedOnSimpleClassGenerator);
+            Assert.fail("Expected a compile error, but no compile error occurred");
+        }
+        catch (Exception exception)
+        {
+            Assert.assertEquals("Error in 'model::MyComplexClass': Can't find type 'model::MyClass'", exception.getMessage());
+        }
+    }
+
+    @Test
+    public void testOverridingCoreModel()
+    {
+        GenerationSpecification generationSpecification = this.getTestGenerationSpecification();
+        PureModelContextData.Builder builder = PureModelContextData.newBuilder();
+        builder.addElement(generationSpecification);
+        Class myClass = new Class();
+        myClass.name = "MyClass";
+        myClass._package = "model";
+        builder.addElement(myClass);
+        ModelGenerationFactory factory = ModelGenerationFactory.newFactory(generationSpecification, builder.build());
+        factory.processModelGenerator(simpleClassGenerator);
+        try
+        {
+            factory.validateAndBuildGeneratedModel();
+            Assert.fail("Expected a validation error but no error occurred");
+        }
+        catch (Exception error)
+        {
+            Assert.assertEquals("Generated element 'model::MyClass' of type Class can't override existing element of type Class", error.getMessage());
+        }
```

Comment:
Can we use Assert.assertThrows for this?

Reply:
And, actually, that comment should apply to more than just that one place. It should apply to all the places with that same pattern.